cmake/Modules/GeneratePkgConfig.cmake

```diff
@@ -28,27 +28,39 @@ function(_compile_features_to_gcc_flags _res _features)
 endfunction()
 
 function(_get_target_property_merging_configs _var_name _target_name _propert_name)
-	get_target_property(vals ${_target_name} ${_propert_name})
-	if (NOT vals)
-		set(vals "")
+	get_property(prop_set TARGET ${_target_name} PROPERTY ${_propert_name} SET)
+	if (prop_set)
+		get_property(vals TARGET ${_target_name} PROPERTY ${_propert_name})
+	else()
 		if (CMAKE_BUILD_TYPE)
 			list(APPEND configs ${CMAKE_BUILD_TYPE})
-		elseif()
+		elseif(CMAKE_CONFIGURATION_TYPES)
 			list(APPEND configs ${CMAKE_CONFIGURATION_TYPES})
 		endif()
 		foreach(cfg ${configs})
 			string(TOUPPER "${cfg}" UPPERCFG)
-			get_target_property(mapped_configs ${_target_name} "MAP_IMPORTED_CONFIG_${UPPERCFG}")
+			get_property(mapped_configs TARGET ${_target_name} PROPERTY "MAP_IMPORTED_CONFIG_${UPPERCFG}")
 			if (mapped_configs)
 				list(GET "${mapped_configs}" 0 target_cfg)
 			else()
 				set(target_cfg "${UPPERCFG}")
 			endif()
-			get_target_property(val_for_cfg  ${_target_name} "${_propert_name}_${target_cfg}")
-			if (val_for_cfg)
+			get_property(prop_set TARGET ${_target_name} PROPERTY ${_propert_name}_${target_cfg} SET)
+			if (prop_set)
+				get_property(val_for_cfg TARGET ${_target_name} PROPERTY ${_propert_name}_${target_cfg})
 				list(APPEND vals "$<$<CONFIG:${cfg}>:${val_for_cfg}>")
+				break()
 			endif()
 		endforeach()
+		if (NOT prop_set)
+			get_property(imported_cfgs TARGET ${_target_name} PROPERTY IMPORTED_CONFIGURATIONS)
+			# CMake docs say we can use any of the imported configs
+			list(GET imported_cfgs 0 imported_config)
```

Comment:
It would not be better to prefer `Release` configuration, if it exists in  `IMPORTED_CONFIGURATIONS`? Something like, if `Release` exists in `imported_cfgs`, then use release, otherwise use an item at 0 index.
I'm thinking about this, like everytime is better to use release build, if there will be more values, like RelWithDebInfo or MinSizeRel or something custom.
Or prefer best choice and if not exists, than use 0 index.

Reply:
Or that's too much :)

Reply:
I should probably check what does CMake itself do and copy its logic.

Reply:
CMake takes the first available: https://gitlab.kitware.com/cmake/cmake/-/blob/master/Source/cmTarget.cxx#L2107

Reply:
cmake doesn't prioritize Release config, it only populates `IMPORTED_CONFIGURATIONS` with all available configurations, so example, if at 0 index will be MinSizeRel configuration and at 1 index will be Release configuration, so of course MinSizeRel config will be picked now

Reply:
I meant something like this:
```
list(FIND imported_cfgs "RELEASE" found)
if(found)
  set(imported_cfg "RELEASE")
else()
  list(GET imported_cfgs 0 imported_config)
endif()
```

Reply:
but if not, its ok, i'm happy with this PR, it solves all problems which we found 👌

Reply:
Sorry, don't really understand you. I replied to you that replicating the internal CMake behavior is the best approach, as far as I cab see. From the CMake sources (linked above) I conclude it chooses the first available configuration, and so should do we. Why do you want to prioritize the release configuration? The .pc file may not differ from the Targets.cmake file, and therefore we have to replicate the internal CMake logic here.

Reply:
You are right, replicate internal cmake logic is good idea anyway, I agree, there is no reason to change this logic

Reply:
OK, thanks.